client/src/controllers/scores/scores_controller.js

```diff
@@ -102,6 +109,27 @@ class ScoresController {
 		return badScores;
 	}
 
+	teamIsSelected() {
+		return !this._loading && this._teamNumberList && this._teamNumberList.indexOf(parseInt(this.search)) > -1
+	}
+
+	selectedTeamMatches() {
+		let self = this
+		if (this.teamIsSelected()) {
+			if(self._matches){
+				return self._matches
+			}
+			self.Tournament.teamsMatches(parseInt(this.search)).then(matches => {
+				self._matches = matches
+			})
+			return self._matches

```

Comment:
The point of a promise is to allow async work. If you call it and then immiddiatly try to use its result, it may not be working fast enough (for example, the response didn't return yet from the server) and you would return here an empty array. This function should return a promise. Speaking of which, not all teams would have the same matches. For example there are re-runs, or teams who decided to miss the second round on purpose. Please don't cache the matches